tests/test_beta_1.py

```diff
@@ -0,0 +1,124 @@
+from beta_1 import QuantumKNearestNeighbours as qkn
+from utils import normalise_vector, pad_vector_with_zeros
+
+class TestBeta1(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls) -> None:
+        """
+        Set up the class for testing.
+        """
+        np.random.seed(args.seed)
+        cls.x_train = [np.random.uniform(
+            -args.x_limit, args.x_limit, size=args.x_size) for _ in range(
+                args.n_train)]
+        cls.x_test = [np.random.uniform(
+            -args.x_limit, args.x_limit, size=args.x_size) for _ in range(
+                args.n_test)]
+        for i,xtr in enumerate(cls.x_train):
+            xtr = normalise_vector(xtr)
+            cls.x_train[i] = pad_vector_with_zeros(xtr)
+        for j,xte in enumerate(cls.x_test):
+            xte = normalise_vector(xte)
+            cls.x_test[j] = pad_vector_with_zeros(xte)
+        cls.y_train = np.random.randint(0, args.n_classes, size = args.n_train)
+        cls.y_test = np.random.randint(0, args.n_classes, size = args.n_test)
+        cls.beta_1 = qkn(
+            cls.x_train, cls.x_test, cls.y_train, cls.y_test, args.k
+        )
+        cls.beta_1.compute_test_train_distances()
+        cls.beta_1.compute_closest_vectors_indexes()
+        cls.beta_1.compute_test_predictions()
+    
+    def test_number_of_train_test_distances_is_correct(self)-> None:
+        """
+        Check that for each train instance, its distance with all the 
+        test vectors is computed. 
+        """
+        for dist_train in self.beta_1.test_train_distances:
+            self.assertEqual(
+                len(dist_train), args.n_train)
+
+    def test_number_of_closest_vectors_indexes_is_correct(self)-> None:
+        """
+        Check that for each train instance, the length of the list 
+        with the closest vectors indexes has length equal to k. 
+        """
+        for indexes in self.beta_1.closest_vectors_indexes:
+            self.assertEqual(
+                len(indexes), args.k
+            )
+
+    def test_save_and_load_train_test_distances(self)-> None:
+        """
+        Check usage of save and load train_test_distances, by ensuring
+        that the distances have the same value before and after saving.
+        """
+        distances_before_saving = self.beta_1.test_train_distances
+        self.beta_1.save_test_train_distances(
+            'test_train_distances', './'
+        )
+        self.beta_1.load_test_train_distances(
+            'test_train_distances.pickle', './'
+        )
+        self.assertEqual(
+            distances_before_saving, self.beta_1.test_train_distances
+        )
+
+    def test_predictions_are_integers(self)-> None:
+        """
+        Check that the predictions output by the model are integers.
+        """
+        for pred in self.beta_1.test_predictions:
+            self.assertIs(type(pred), np.int64)
+            
+
+if __name__ == '__main__':
```

Comment:
Are we sure we want a main in a test file?

Reply:
You can see how a main is used in the docs of unittest : 

https://docs.python.org/3/library/unittest.html#basic-example

Reply:
Ok, sorry, what I said wasn't very clear. Let me rephrase my question: why do you have such an extensive main with argument parsing for so many parameters?
Are you typing them in the command line every time you run a test? Or did you add them in case someone wanted to parameterise the tests and you use the default every time? or some other way of doing things?
I looked for a note on this in the readme but didn't find one, maybe it's there and I didn't see it.

Reply:
The parameters are there with the aim of not hardcoding parameters, and make things generic if possible. It was one of the comments in previous PRs. 

You are not typing them every time you run a test. You can see there how a default value is specified, which is going to be used if the user don't specify anything in the command line when running the test file. It is exactly the same way as we use to do it in the pipelines (use_{model}.py) for the report.

Reply:
I see where this all came from now!
I've set up a quick meeting to discuss this on Monday :).